src/java.base/share/classes/java/lang/invoke/X-VarHandleSegmentView.java.template

```diff
@@ -108,39 +108,34 @@ final class VarHandleSegmentAs$Type$s extends VarHandleSegmentViewBase {
         if ((alignmentMask & NON_PLAIN_ACCESS_MIN_ALIGN_MASK) != NON_PLAIN_ACCESS_MIN_ALIGN_MASK) {
             throw VarHandleSegmentViewBase.newUnsupportedAccessModeForAlignment(alignmentMask + 1);
         }
-        return offsetPlain(bb, offset, alignmentMask);
+        return offsetPlain(bb, offset);
     }
 
     @ForceInline
-    static long offsetPlain(AbstractMemorySegmentImpl bb, long offset, long alignmentMask) {
+    static long offsetPlain(AbstractMemorySegmentImpl bb, long offset) {
         long base = bb.unsafeGetOffset();
-        long address = base + offset;
-        long maxAlignMask = bb.maxAlignMask();
-        if (((address | maxAlignMask) & alignmentMask) != 0) {
-            throw VarHandleSegmentViewBase.newIllegalArgumentExceptionForMisalignedAccess(address);
-        }
-        return address;
+        return base + offset;
     }
 
     @ForceInline
     static $type$ get(VarHandle ob, Object obb, long base) {
         VarHandleSegmentViewBase handle = (VarHandleSegmentViewBase)ob;
-        AbstractMemorySegmentImpl bb = checkAddress(obb, base, handle.length, true);
+        AbstractMemorySegmentImpl bb = checkReadOnly(obb, true);
```

Comment:
For getter methods, which pass a constant `true` here, `checkReadOnly` essentially just does a null check and cast on the segment. Not sure if it's worth simplifying... (I'm happy if you want to leave it like this as well)

Reply:
I'll leave it for now. There's always a trade-off with these generated templates...